apps/omg_child_chain/lib/omg_child_chain/child_manager.ex

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.ChildManager do
+  @moduledoc """
+    Reports it's health to the Monitor after start or restart and shutsdown.
+  """
+  use GenServer, restart: :transient
+
+  require Logger
+  @timer 5_000
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  def init(args) do
+    monitor = Keyword.fetch!(args, :monitor)
+    {:ok, _tref} = :timer.send_after(@timer, :health_checkin)
```

Comment:
Curious why we need to delay the checkin, and delay by 5s?

Reply:
no technical reason! reduced it to 100miliseconds. I would expand the manager if we see it beneficial, so that processes on the same level checkin to him and once they're all checked it, it would forward the checkin to monitor. for now it works OK.